tools/toolchain/optimized_clang.sh

```diff
@@ -156,3 +156,6 @@ fi
 cd "${CLANG_BUILD_DIR}"
 ninja -C build install-distribution-stripped
 dnf remove -y clang clang-libs
+# above package removal might have removed those symbolic links, which will cause ccache not to work later on. Manually restore them.
+ln -s /usr/bin/ccache /usr/lib64/ccache/clang || true
+ln -s /usr/bin/ccache /usr/lib64/ccache/clang++ || true
```

Comment:
```console
$ dnf whatprovides /usr/lib64/ccache/clang
Last metadata expiration check: 8 days, 23:58:17 ago on Sat 31 Aug 2024 02:25:56 PM CST.
ccache-4.8.3-3.fc40.x86_64 : C/C++ compiler cache
Repo        : fedora
Matched from:
Filename    : /usr/lib64/ccache/clang

ccache-4.9.1-1.fc40.x86_64 : C/C++ compiler cache
Repo        : @System
Matched from:
Filename    : /usr/lib64/ccache/clang

ccache-4.9.1-1.fc40.x86_64 : C/C++ compiler cache
Repo        : updates
Matched from:
Filename    : /usr/lib64/ccache/clang
```

so, removing `clang` or `clang-libs` does not nuke `/usr/lib64/ccache/clang`

> Does is work for aarch64 (does it use /usr/lib64 or /usr/lib?)

yes. it does. see https://fedora.pkgs.org/40/fedora-aarch64/ccache-4.8.3-3.fc40.aarch64.rpm.html . but i think we should probably rely on `ccache` package to provide these symbolic links.